src/libcore/cell.rs

```diff
@@ -280,6 +281,19 @@ unsafe impl<T> Send for Cell<T> where T: Send {}
 #[stable(feature = "rust1", since = "1.0.0")]
 impl<T> !Sync for Cell<T> {}
 
+#[unstable(feature = "move_cell", issue = "39264")]
+impl<T:Clone> Clone for Cell<T> {
+    #[inline]
+    default fn clone(&self) -> Cell<T> {
+        let temp = unsafe { mem::uninitialized() };
+        let inner = self.replace(temp);
+        // If `clone` panics, let it crash.
```

Comment:
1. `AssertUnwindSafe` is not `unsafe`, and for good reason. Safe code using `AssertUnwindSafe` unwisely may get very undesirable behavior, but not UB.
2. `AssertUnwindSafe` is irrelevant, as this problem occurs without any `catch_unwind` calls. In fact, a well-placed `catch_unwind` can *stop* a panic in this `clone` call from dropping `*self` (by stopping the unwinding before it gets to the stack frame where the `Cell` would be dropped).